src/AsyncTCP.cpp

```diff
@@ -1614,7 +1620,11 @@ int8_t AsyncServer::_accept(tcp_pcb *pcb, int8_t err) {
     AsyncClient *c = new AsyncClient(pcb);
     if (c) {
       c->setNoDelay(_noDelay);
-      return _tcp_accept(this, c);
+      const int8_t err = _tcp_accept(this, c);
+      if (err != ERR_OK) {
+        delete c;
+      }
+      return err;
```

Comment:
if the accept event cannot be added to the queue then delete the allocated client class.